CIP-0118/README-implicit.md

```diff
@@ -0,0 +1,937 @@
+3. `requiredTxs : TxId ↦ B`, which contains all the IDs of the transactions that have to be included 
+in the zone, associated to a boolean value that specifies whether the full `TxInfo` value for the 
+transaction with a given `txId` is exposed to Plutus scripts that are executed as part of validating 
+that transaction 
+
+The field `requiredFullTxBodies : Tx` is included in `Tx` to enable passing the full transaction data 
+to Plutus scripts. This field is computed at the time of deserialization to contain all transactions 
+corresponding to each `(txId, True)` in `requiredTxs`, with the `requiredTxs` field instantiated to `{}`.
+Possibly, this field should be of type `TxInfo` instead.
+
+### TxInfo
+
+The `TxInfo` shown to a Plutus script is replaced with `(TxInfo, List TxInfo)` where the first
+element of the pair is the usual `TxInfo`, including fields (1) and (2), the missing and extra value. 
+The second element is a list of `TxInfo` values for all the transactions for which `TxInfo` (or `Tx`) is included 
+in the `requiredFullTxBodies` field.
+
+
+### Ledger Rule Changes
+
+The prototype for the changes can be found
+[here](https://github.com/IntersectMBO/formal-ledger-specifications/compare/polina-babel?expand=1)
+
+#### ZONE
+
+Currently, the top-level transaction-processing transition system in the ledger system is `LEDGERS`.
+It considers a list of transactions in the block body valid whenever `LEDGER` is valid for
+each transaction, checked in the order they appear in that list. We propose replacing `LEDGERS`
+with
+
+`ZONES ⊆ LEnv x LState x List (List Tx) x LState`
+
+as the top level transition, which processes each zone in the block by
+validating it with the transition
+
+`ZONE ⊆ LEnv x LState x List Tx x LState`
+
+The `ZONE` system includes two rules : `ZONE-V` may be applicable when all transactions are tagged with
+`isValid` tag that is `True`, and `ZONE-N` may be applicable otherwise. The reason
+for differentiating phase-2 valid and invalid _zones_ (and not just at the level
+of individual transactions) is that it may not be possible to apply subsequent
+transactions in a zone once a single transaction is found to be phase-2 invalid.
+Recall that  
+it is not possible to exclude only _some_ transactions from a zone in a general way,
+so, unlike the existing design, the mempool is no able to reject individual zone transactions
+dependent on a preceding transaction being phase-2 valid. We design the `ZONE-N`
+rule to deal with this situation in a special way. 
+
+The following checks are done by both rules :
+
+1. the sum total of the sizes of all transactions in the zone is less than the `maxTxSize`
+protocol parameter (see [Network Requirements and Changes](#network-requirements-and-changes)), and
+2. the collateral provided by each transaction `tx` for script execution is sufficient
+to cover **the size-based fees for all transactions corresponding to transactions in `requiredTxs`**
+(see [DDoS](#ddos))
```

Comment:
Providing collateral fro all preceding transactions made sense to me. This change that everyone specifies their own collateral raises this questions:

say we have theses txs in the zone: `[tx1, tx2, tx3, tx4]`

While the nodes validates all these transactions in this sequence and only `tx3` fails phase2 validation then is the collateral collected for all of the transactions in the zone or just the ones that have already been validated or just the one that failed?

None of the decisions who pays the collateral is fair as far as I can understand it.

Reply:
As a node that runs the babel-fee service, you get a transaction with a failing script, you put that on-chain with its `requiredTxs` purely for collateral collection, as a zone. If it was meant to complete an existing zone, that zone is still incomplete, and awaiting other transactions to complete it. However, you've already validated its scripts, but are waiting on new (valid) transactions to complete the zone, you don't need to re-validate them. 

As a node that is simply processing blocks, you don't even validate scripts in any transactions in a phase-2 invalid zone *except* for the single phase-2 invalid tx, whose scripts you check to make sure they do indeed fail.

Reply:
cardano-node does not know if the zone or any of the transactions in the zone are phase2 invalid! phase2 validity flag is ignored by the block producing node!

From Alonzo spec:
> In the implementation the IsValid tag is part of a user-submitted transaction, but it is ignored and re-computed by the block producer.

So, the node will have to execute all of the scripts in the zone, but it will only get some of the collateral with this approach. That is my concern that I am trying to explain.